### ISSUE TITLE
Allow for criteria sort precedence using array

### ISSUE THREAD

**Issue by dzotcrew** [balderdashy/waterline#694]
Case
Sails model:
```javascript
module.exports = {
    "schema": true,
    "tableName": "orders",
    "autoCreatedAt": false,
    "autoUpdatedAt": false,
    "attributes": {
        "id": {
            "columnName": "order_id_pk",
            "required": true,
            "type": "integer",
            "unique": true,
            "primaryKey": true
        },
        "positionId": {
            "columnName": "position_id_fk",
            "required": true,
            "type": "integer"
        },
        "description": {
            "columnName": "description",
            "type": "text"
        },
		....
    }
};
```
Request from angular sails:
```javascript
$sails.get('/api/order', {
	sort: {
	  positionId: 1,
	  description: -1
	}
});
```
Expected behavior:
```sql
...ORDER BY position_id_fk ASC, description DESC
```
But after Transformation.serialize method this request transform to:
```sql
... ORDER BY description DESC, position_id_fk ASC
```

Because in waterline/lib/waterline/core/transformations.js:121 positionId after transformation is added to the end of the sorting criteria, and a description(because it isn't transformed) retains its position.

**Comment by dmarcelino** [comment]
:+1: to:
``` javascript
    sort: [
      { positionId: 1 },
      { description: -1 }
    ]
```

Unfortunately this will need to be addressed on the adapters so we're talking about a new API, potentially. @dzotcrew would you be able to submit a test to [waterline-adapter-tests](https://github.com/balderdashy/waterline-adapter-tests/blob/master/interfaces/queryable/modifiers/sort.modifier.test.js)?

It would also be good to see what other people say.

**Comment by atiertant** [comment]
sort: [
      { positionId: 1 },
      { description: -1 }
    ]

i thought that's not elegant although it's working, why don't use : 

    sort: [ 'positionId ASC' , 'description DESC' ]

**Comment by dmarcelino** [comment]
Given `sort` is supported in the following forms (check [sort.modifier.test.js](https://github.com/balderdashy/waterline-adapter-tests/blob/master/interfaces/queryable/modifiers/sort.modifier.test.js)):
``` javascript
sort: { dob: 1 }
sort: 'dob desc' 
```

I believe we will need to support both `sort: [ { positionId: 1 },  { description: -1 } ]` and `sort: [ 'positionId ASC' , 'description DESC' ]` to remain consistent.

**Comment by particlebanana** [comment]
Would be fairly trivial to add this and make it backwards compatible. Can't believe this hasn't come up more often. I guess most of the time we are creating these sort objects on the server where V8 keeps the order. We shouldn't rely on that though.

**Comment by sailsbot** [comment]
Thanks for posting, @dzotcrew.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by mactyr** [comment]
Can you please confirm whether this was addressed, and if so what the recommended way of sorting on multiple attributes/columns is now? The [documentation](http://sailsjs.com/documentation/reference/waterline-orm/queries/sort) only shows an example with one attribute/column, and I tried some of the syntaxes under discussion above but none of them seem to be working for me. (Passing sort() a single object with a couple of properties _is_ working for me and I'm on node where it sounds like that is safe for now, but it would be good to know the right way.) Thanks!
